pkg/providers/imagefamily/bootstrap/parser_helpers.go

```diff
@@ -0,0 +1,684 @@
+		"getMaxLBRuleCount":                         getMaxLBRuleCount,
+		"getGpuImageSha":                            getGpuImageSha,
+		"getGpuDriverVersion":                       getGpuDriverVersion,
+		"getIsSgxEnabledSKU":                        getIsSgxEnabledSKU,
+		"getShouldConfigureHTTPProxy":               getShouldConfigureHTTPProxy,
+		"getShouldConfigureHTTPProxyCA":             getShouldConfigureHTTPProxyCA,
+		"getAzureEnvironmentFilepath":               getAzureEnvironmentFilepath,
+		"getLinuxAdminUsername":                     getLinuxAdminUsername,
+		"getTargetEnvironment":                      getTargetEnvironment,
+		"getIsVHD":                                  getIsVHD,
+		"getDisableSSH":                             getDisableSSH,
+		"getServicePrincipalFileContent":            getServicePrincipalFileContent,
+		"getEnableSwapConfig":                       getEnableSwapConfig,
+		"getShouldCOnfigTransparentHugePage":        getShouldCOnfigTransparentHugePage,
+		"getProxyVariables":                         getProxyVariables,
+		"getHasKubeletDiskType":                     getHasKubeletDiskType,
+		"getInitAKSCustomCloudFilepath":             getInitAKSCustomCloudFilepath,
+		"getTargetCloud":                            getTargetCloud,
+		"getIsAksCustomCloud":                       getIsAksCustomCloud,
+		"getGPUNeedsFabricManager":                  getGPUNeedsFabricManager,
+		"getEnableNvidia":                           getEnableNvidia,
+	}
+}
+
+func getFuncMapForContainerdConfigTemplate() template.FuncMap {
+	return template.FuncMap{
+		"derefBool":                        deref[bool],
+		"getIsKrustlet":                    getIsKrustlet,
+		"getEnsureNoDupePromiscuousBridge": getEnsureNoDupePromiscuousBridge,
+		"isKubernetesVersionGe":            nbcontractv1.IsKubernetesVersionGe,
+		"getHasDataDir":                    getHasDataDir,
+		"getEnableNvidia":                  getEnableNvidia,
+	}
+}
+
+func getStringFromVMType(enum nbcontractv1.ClusterConfig_VM) string {
+	switch enum {
+	case nbcontractv1.ClusterConfig_STANDARD:
+		return nbcontractv1.VMTypeStandard
+	case nbcontractv1.ClusterConfig_VMSS:
+		return nbcontractv1.VMTypeVmss
+	case nbcontractv1.ClusterConfig_UNSPECIFIED:
+		return ""
+	default:
+		return ""
+	}
+}
+
+//nolint:exhaustive // NetworkPlugin_NP_NONE and NetworkPlugin_NP_UNSPECIFIED should both return ""
+func getStringFromNetworkPluginType(enum nbcontractv1.NetworkPlugin) string {
+	switch enum {
+	case nbcontractv1.NetworkPlugin_NP_AZURE:
+		return nbcontractv1.NetworkPluginAzure
+	case nbcontractv1.NetworkPlugin_NP_KUBENET:
+		return nbcontractv1.NetworkPluginKubenet
+	default:
+		return ""
+	}
+}
+
+//nolint:exhaustive // NetworkPolicy_NPO_NONE and NetworkPolicy_NPO_UNSPECIFIED should both return ""
+func getStringFromNetworkPolicyType(enum nbcontractv1.NetworkPolicy) string {
+	switch enum {
+	case nbcontractv1.NetworkPolicy_NPO_AZURE:
```

Comment:
What does the O stand for here?

Reply:
protobuf 3 doesn't allow us to have same name in 2 groups of enum.
```
enum NetworkPlugin {
  NP_UNSPECIFIED = 0;
  NP_NONE = 1;
  NP_AZURE = 2;
  NP_KUBENET = 3;
}

enum NetworkPolicy {
  NPO_UNSPECIFIED = 0;
  NPO_NONE = 1;
  NPO_AZURE = 2;
  NPO_CALICO = 3;
}
```
that means we can't have `NP_AZURE` in both `NetworkPlugin` and `NetworkPolicy`. That's why I put PO, which stands for **PO**licy to distinguish them.